src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1470,15 +1469,18 @@ private ReadOnlySpan<uint> ToUInt32Span(Span<uint> scratch)
             }
 
             // Find highest significant byte and ensure high bit is 0 if positive, 1 if negative
-            int msb;
-            for (msb = dwords.Length - 1; msb > 0 && dwords[msb] == highDWord; msb--);
+            int msb = dwords.Length - 1;
+            while (msb > 0 && dwords[msb] == highDWord)
+            {
+                msb--;
+            }
```

Comment:
Why is this better?

Reply:
I don't have a strong preference here, but I've never personally been a fan of `for (...) ;` style blocks, as they are very easy to misread or accidentally break during refactorings (it's the same reason why I always use braces, even for single line statements).

This should generate the same code and makes the logic clearer/more explicit.